### PR TITLE
Test about student login and logout

### PR DESCRIPTION
**Pull request by henrywang1**
1. Test about student login and logout
2. Test environment setup

- note: env.rb is generated by cucumber

### CODE REVIEW

File: zlp-scheduler/features/step_definitions/authentication_steps.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Before do
+  # The current sessions_controller requires at least one active term
+  @term = FactoryBot.create(:term, :active=>true)
+  @term.save()
+end
```
Comment (BWitemeyer) [inline zlp-scheduler/features/step_definitions/authentication_steps.rb:5]: I just want to make sure I'm understanding what this does correctly. Before anything test-wise is done, it creates an active term in the db? Is that correct?

Comment (henrywang1) [reply zlp-scheduler/features/step_definitions/authentication_steps.rb:5]: Yes, you are correct. The "before" block will be run before every scenario.

**Comment by BWitemeyer** [comment]
I think this looks great!

**Comment by MinhDang98** [comment]
This looks good Han-Yi

**Review by MinhDang98 — APPROVED** [review]

**Review by MinhDang98 — COMMENTED** [review]
Well done man!

File: zlp-scheduler/spec/factories.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    factory :cohort do
+        id {rand(10000)}
+    end
+
+    factory :user do
+        firstname {Faker::Name.name}
+        lastname {Faker::Name.name}
+        email {Faker::Internet.email}
+        password {Faker::Internet.password}
+    end
```
Comment (adauzat90) [inline zlp-scheduler/spec/factories.rb:11]: Does this factory create valid users? For a user to be valid they have to have @tamu.edu at then of the email.

Comment (henrywang1) [reply zlp-scheduler/spec/factories.rb:11]: That's a good point. I will update it, and add another scenario about user login with non-tamu email.

**Pull request merged**